### PR TITLE
Signup route

### PR DESCRIPTION
**Pull request by hashemt**
Relates #59

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +41,8 @@
     "cookie-parser": "^1.4.4",
     "dotenv": "^8.1.0",
     "express": "^4.17.1",
+    "express-fileupload": "^1.1.6-alpha.6",
```
Comment (AbdallahAmmar96) [inline package.json]: delete this module

File: server/app.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-
 const express = require('express');
+const formData = require('express-form-data');
 require('dotenv').config();
 
 const router = require('./router');
 
 const PORT = process.env.PORT || 5000;
 const app = express();
+app.use(formData.parse());
```
Comment (AbdallahAmmar96) [inline server/app.js]: move it down

File: client/src/components/pages/Signup/index.js [hunk 3]
```diff
@@ -35,14 +36,20 @@ class SignupForm extends Component {
...
         const formData = new FormData();
         const data = { ...values };
-        data[remote] = remote;
-        formData.append('data', data);
-        formData.append('avalibility', available);
-        formData.append('image', this.image);
+        data.remote = remote;
+        formData.append('data', JSON.stringify(data));
+        formData.append('avalibility', JSON.stringify(available));
+        const files = this.image.state.fileList[0].originFileObj;
```
Comment (AbdallahAmmar96) [inline client/src/components/pages/Signup/index.js]:
```suggestion
        const file = this.image.state.fileList[0].originFileObj;
```

File: client/src/components/pages/Signup/index.js [hunk 4]
```diff
@@ -35,14 +36,20 @@ class SignupForm extends Component {
...
-    validateFieldsAndScroll((err, values) => {
+    validateFieldsAndScroll(async (err, values) => {
       if (!err) {
         const formData = new FormData();
         const data = { ...values };
-        data[remote] = remote;
-        formData.append('data', data);
-        formData.append('avalibility', available);
-        formData.append('image', this.image);
+        data.remote = remote;
```
Comment (MohammadAlhalaq) [inline client/src/components/pages/Signup/index.js]:
```suggestion
        data.remote = remote;
        data.set('remote', remote)
```

File: server/controllers/routes/signup.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    const avalibility = JSON.parse(req.body.avalibility);
+    let imageUrl = '';
+    if (req.files) {
+      const { image } = req.files;
+      await imageSchema.validate({ image });
+      const uploadedImage = await cloudinary.uploader.upload(image.path);
+      imageUrl = uploadedImage.url;
+    }
+    await signupSchema.validate({ ...therapist, avalibility });
+    const hashedPassword = await hashPassowrd(therapist.password);
```
Comment (MohammadAlhalaq) [inline server/controllers/routes/signup.js]:
```suggestion
    const hashedPassword = await bcrypt.hash(therapist.password, 10);
```

File: server/controllers/routes/signup.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+const { hashPassowrd } = require('../utils/hashingPassword');
+const { signupSchema, imageSchema } = require('../validation');
+const base = require('../../database/config/airtableConnection');
+const cloudinary = require('../../database/config/cloudinaryConnection');
+
+module.exports = async (req, res, next) => {
+  try {
```
Comment (MohammadAlhalaq) [inline server/controllers/routes/signup.js]:
```js
    const { data, availability } = req.body;
```

File: client/src/components/pages/Signup/index.js [hunk 7]
```diff
@@ -93,6 +100,15 @@ class SignupForm extends Component {
     const {
       form: { getFieldDecorator },
     } = this.props;
+    const languages = ['arabic', 'english', 'spanish', 'french'];
```
Comment (AbdallahAmmar96) [inline client/src/components/pages/Signup/index.js]: move this to staticData file

File: client/src/components/pages/Signup/index.js [hunk 8]
```diff
@@ -93,6 +100,15 @@ class SignupForm extends Component {
...
+    const prices = [
```
Comment (AbdallahAmmar96) [inline client/src/components/pages/Signup/index.js]: To the static file

File: client/src/components/pages/Signup/index.js [hunk 9]
```diff
@@ -201,10 +217,9 @@ class SignupForm extends Component {
               ],
             })(
               <Select placeholder="Select a range" onChange={this.handlePrice}>
-                <Option value="20-30">20-30</Option>
-                <Option value="30-40">30-40</Option>
-                <Option value="40-50">40-50</Option>
-                <Option value="50-60">50-60</Option>
+                {prices.map(price => (
+                  <Option value={price}>{price}</Option>
```
Comment (AbdallahAmmar96) [inline client/src/components/pages/Signup/index.js]: you need to add a key

File: client/src/components/pages/Signup/index.js [hunk 10]
```diff
@@ -221,6 +236,59 @@ class SignupForm extends Component {
               ],
             })(<Input placeholder="Enter post code" />)}
           </Form.Item>
+          <Form.Item label="Languages spoken:">
+            {getFieldDecorator('language', {
+              rules: [
+                {
+                  required: true,
+                  message: 'Please input your language!',
```
Comment (AbdallahAmmar96) [inline client/src/components/pages/Signup/index.js]:
```suggestion
                  message: 'Please select your languages!',
```

File: client/src/components/pages/Signup/index.js [hunk 11]
```diff
@@ -221,6 +236,59 @@ class SignupForm extends Component {
...
+                },
+              ],
+            })(
+              <Select
+                mode="multiple"
+                style={{ width: '100%' }}
+                placeholder="select one country"
```
Comment (AbdallahAmmar96) [inline client/src/components/pages/Signup/index.js]:
```suggestion
                placeholder="Select your country"
```

File: client/src/components/pages/Signup/index.js [hunk 12]
```diff
@@ -221,6 +236,59 @@ class SignupForm extends Component {
...
+                {
+                  message: 'The approch is not valid!',
+                },
+                {
+                  required: true,
+                  message: 'Please input your approch!',
+                },
+                {
+                  max: 200,
+                  message: 'should be 200 words as a max',
```
Comment (AbdallahAmmar96) [inline client/src/components/pages/Signup/index.js]:
```suggestion
                  message: 'Maximum 200 characters',
```

**Pull request merged**